IdentityCore/src/broker_operation/request/account_request/MSIDBrokerOperationRemoveAccountRequest.m

```diff
@@ -0,0 +1,93 @@
+//------------------------------------------------------------------------------
+//
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//------------------------------------------------------------------------------
+
+#import "MSIDBrokerOperationRemoveAccountRequest.h"
+#import "MSIDJsonSerializableFactory.h"
+#import "NSDictionary+MSIDExtensions.h"
+#import "MSIDAccountIdentifier.h"
+#import "MSIDJsonSerializableTypes.h"
+#import "MSIDConstants.h"
+
+@implementation MSIDBrokerOperationRemoveAccountRequest
+
++ (void)load
+{
+    [MSIDJsonSerializableFactory registerClass:self forClassType:self.operation];
+}
+
+#pragma mark - MSIDBrokerOperationRequest
+
++ (NSString *)operation
+{
+    return MSID_JSON_TYPE_OPERATION_REQUEST_REMOVE_ACCOUNT;
+}
+
+#pragma mark - MSIDJsonSerializable
+
+- (instancetype)initWithJSONDictionary:(NSDictionary *)json error:(NSError **)error
+{
+    self = [super initWithJSONDictionary:json error:error];
+    
+    if (self)
+    {
+        _accountIdentifier = [[MSIDAccountIdentifier alloc] initWithJSONDictionary:json error:error];
+        if (!_accountIdentifier || !_accountIdentifier.homeAccountId)
```

Comment:
I still think there might be cases where home account id is not available. Let's discuss offline.

Reply:
Discussed offline. We will keep it as is for now. An item is created to keep track of this issue: https://github.com/AzureAD/microsoft-authentication-library-for-objc/issues/762